src/services/expressions.ts

```diff
@@ -21,37 +20,30 @@ import { getFieldsVariable, getLabelsVariable } from './variableGetters';
  * @param excludeEmpty - if true, the query will exclude empty values for the given streamSelectorName
  */
 export function getTimeSeriesExpr(sceneRef: SceneObject, streamSelectorName: string, excludeEmpty = true): string {
-  const labelsVariable = getLabelsVariable(sceneRef);
   const fieldsVariable = getFieldsVariable(sceneRef);
 
-  let labelExpressionToAdd;
   let metadataExpressionToAdd = '';
   if (excludeEmpty) {
     // `LEVEL_VARIABLE_VALUE` is a special case where we don't want to add this to the stream selector
-    if (streamSelectorName !== LEVEL_VARIABLE_VALUE) {
-      labelExpressionToAdd = { key: streamSelectorName, operator: '!=', value: '' };
-    } else {
+    if (streamSelectorName === LEVEL_VARIABLE_VALUE) {
       metadataExpressionToAdd = `| ${LEVEL_VARIABLE_VALUE} != ""`;
     }
   }
 
-  const labelFilters = [...labelsVariable.state.filters, labelExpressionToAdd].filter(isDefined);
-  const streamSelectors = renderLogQLLabelFilters(labelFilters);
-
   const fieldFilters = fieldsVariable.state.filters;
   const parser = getParserFromFieldsFilters(fieldsVariable);
 
   // if we have fields, we also need to add parsers
   if (fieldFilters.length) {
     if (parser === 'mixed') {
-      return `sum(count_over_time({${streamSelectors}} ${metadataExpressionToAdd} ${VAR_LINE_FILTER_EXPR} ${VAR_PATTERNS_EXPR} ${MIXED_FORMAT_EXPR} ${VAR_FIELDS_EXPR} [$__auto])) by (${streamSelectorName})`;
+      return `sum(count_over_time({${VAR_LABELS_EXPR}} ${metadataExpressionToAdd} ${VAR_LINE_FILTER_EXPR} ${VAR_PATTERNS_EXPR} ${MIXED_FORMAT_EXPR} ${VAR_FIELDS_EXPR} [$__auto])) by (${streamSelectorName})`;
     }
     if (parser === 'json') {
-      return `sum(count_over_time({${streamSelectors}} ${metadataExpressionToAdd} ${VAR_LINE_FILTER_EXPR} ${VAR_PATTERNS_EXPR} ${JSON_FORMAT_EXPR} ${VAR_FIELDS_EXPR} [$__auto])) by (${streamSelectorName})`;
+      return `sum(count_over_time({${VAR_LABELS_EXPR}} ${metadataExpressionToAdd} ${VAR_LINE_FILTER_EXPR} ${VAR_PATTERNS_EXPR} ${JSON_FORMAT_EXPR} ${VAR_FIELDS_EXPR} [$__auto])) by (${streamSelectorName})`;
     }
     if (parser === 'logfmt') {
-      return `sum(count_over_time({${streamSelectors}} ${metadataExpressionToAdd} ${VAR_LINE_FILTER_EXPR} ${VAR_PATTERNS_EXPR} ${LOGS_FORMAT_EXPR} ${VAR_FIELDS_EXPR} [$__auto])) by (${streamSelectorName})`;
+      return `sum(count_over_time({${VAR_LABELS_EXPR}} ${metadataExpressionToAdd} ${VAR_LINE_FILTER_EXPR} ${VAR_PATTERNS_EXPR} ${LOGS_FORMAT_EXPR} ${VAR_FIELDS_EXPR} [$__auto])) by (${streamSelectorName})`;
     }
   }
-  return `sum(count_over_time({${streamSelectors}} ${metadataExpressionToAdd} ${VAR_LINE_FILTER_EXPR} ${VAR_PATTERNS_EXPR} [$__auto])) by (${streamSelectorName})`;
+  return `sum(count_over_time({${VAR_LABELS_EXPR}} ${metadataExpressionToAdd} ${VAR_LINE_FILTER_EXPR} ${VAR_PATTERNS_EXPR} ${VAR_FIELDS_EXPR} [$__auto])) by (${streamSelectorName})`;
```

Comment:
Was this missing? What's the context of this change?

Reply:
Yeah this was missing, if you had structured metadata added for another field besides the current breakdown it wouldn't add it to the query if we hadn't added a parser yet

Reply:
Fantastic, thanks.